citations-impl/impl/src/java/org/sakaiproject/citation/impl/soloapi/JournalConverter.java

```diff
@@ -0,0 +1,45 @@
+package org.sakaiproject.citation.impl.soloapi;
+
+import org.sakaiproject.citation.api.Citation;
+
+import java.util.ArrayList;
+
+/**
+ * Converts from a Context to a Citation.
+ * @author nickwilson
+ *
+ */
+public class JournalConverter extends AbstractConverter  {
+
+	public String getType() {
+		return "JOUR";
+	}
+
+	@Override
+	public Citation convert(ContextObject context) {
+		Citation citation = super.convert(context);
+
+		setCitationProperty(citation, primoNMBibNode, "creator", new ArrayList<String>(){{  add("record"); add("addata"); add("au");}} );
```

Comment:
This will create lots of heap allocations/collections because of all the new arraylists.